### ISSUE TITLE
Execute-ProcessAsUser seems to not always starting a process

### ISSUE THREAD

**Issue by FRacine** [PSAppDeployToolkit/PSAppDeployToolkit#318]
We have several scenario using Execute-ProcessAsUser. Most of them are using -wait and in those scenarios, we do not have problems as the process is starting. Something it may take minutes to start the process but I never hear an issue about process not starting.

However, on process not using -wait, sometimes, the process is just not starting from time to time.

```
		[psobject]$schTaskResult = Execute-Process -Path $exeSchTasks -Parameters "/run /i /tn $schTaskName" -WindowStyle 'Hidden' -CreateNoWindow -Passthru
		If ($schTaskResult.ExitCode -ne 0) {
			[int32]$executeProcessAsUserExitCode = $schTaskResult.ExitCode
			Write-Log -Message "Failed to trigger scheduled task [$schTaskName]." -Severity 3 -Source ${CmdletName}
			#  Delete Scheduled Task
			Write-Log -Message 'Delete the scheduled task which did not trigger.' -Source ${CmdletName}
			Execute-Process -Path $exeSchTasks -Parameters "/delete /tn $schTaskName /f" -WindowStyle 'Hidden' -CreateNoWindow -ContinueOnError $true
			If (-not $ContinueOnError) {
				Throw "Failed to trigger scheduled task [$schTaskName]."
			}
			Else {
				Return
			}
```
From what I see the script is not waiting to start the process so it is possible if the process is too slow the schtaskname might be remove before the task would start.

How may we solve this issue?

Thanks,

**Comment by Latupho** [comment]
Had the same issue.
If you go in the main.ps1 and stop PSADTK from deleting the scheduled task, and edit the task to not as the highest privilege, it will run fine.

It's a scheduled task issue, as of what I have found out.

**Comment by FRacine** [comment]
Hi,

I am happy someone got the same issue but your solution is unclear for me. Would you suggest how improving the code to solve that issue?

Thanks,

**Comment by seanlillis** [comment]
@Latupho, can you clarify why changing the task not to run as highest privilege resolves the issue?

**Comment by seanlillis** [comment]
This will be resolved in the next release, duplicate of #281

**Comment by seanlillis** [comment]
This is fixed in v3.7.0.
